core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,260 @@
+  KOKKOS_INLINE_FUNCTION constexpr BasicView(const BasicView &) = default;
+  KOKKOS_INLINE_FUNCTION constexpr BasicView(BasicView &&) noexcept = default;
+
+  KOKKOS_INLINE_FUNCTION constexpr BasicView &
+  operator=(const BasicView &) = default;
+
+  KOKKOS_INLINE_FUNCTION constexpr BasicView &
+  operator=(BasicView &&) = default;
+
+  ///
+  /// Converting constructor
+  ///
+  template <class OtherElementType, class OtherExtents,
+            class OtherExecutionSpace, class OtherMemorySpace,
+            class OtherLayoutPolicy,
+            typename = std::enable_if_t<std::is_constructible_v<
+                mdspan_type,
+                const mdspan<OtherElementType, OtherExtents,
+                             typename Impl::kokkos_layout_conversion<OtherLayoutPolicy>::type,
+                             default_accessor<OtherElementType>> &>>>
+  KOKKOS_INLINE_FUNCTION constexpr BasicView(
+      const BasicView<OtherElementType, OtherExtents, OtherExecutionSpace,
+                      OtherMemorySpace, OtherLayoutPolicy> &other)
+      : m_tracker(other.m_tracker), m_data(other.m_data) {}
+
+  // TODO: subview constructor
+
+  ///
+  /// Construct from a given mapping
+  ///
+  KOKKOS_INLINE_FUNCTION explicit constexpr BasicView(
+      const std::string &label, const mapping_type &mapping)
+      : BasicView(mapping, label, default_alloc_params{}) {}
+
+  ///
+  /// Construct from a given extents
+  ///
+  KOKKOS_INLINE_FUNCTION explicit constexpr BasicView(const std::string &label,
+                                                      const extents_type &ext)
+      : BasicView(label, mapping_type{ext}) {}
+
+  template <typename... OtherIndexTypes>
+  KOKKOS_INLINE_FUNCTION explicit constexpr BasicView(
+      const std::enable_if_t<
+          std::is_constructible_v<extents_type, OtherIndexTypes...>,
+          std::string> &label,
+      OtherIndexTypes... indices)
+      : BasicView(label, extents_type{indices...}) {}
+  ///@}
+
+  KOKKOS_INLINE_FUNCTION constexpr std::string label() const noexcept {
+    return m_tracker.get_label<memory_space>();
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr extents_type extents() const noexcept {
+    return m_data.extents();
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr index_type extent(
+      rank_type r) const noexcept {
+    return m_data.extent(r);
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr index_type static_extent(
+      rank_type r) const noexcept {
+    return m_data.static_extent(r);
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr index_type size() const noexcept {
+    return m_data.size();
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr index_type stride(
+      rank_type r) const noexcept {
+    return m_data.stride(r);
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr index_type span() const noexcept {
+    return m_data.mapping().required_span_size();
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr bool span_is_contiguous() const noexcept {
+    return m_data.is_exhaustive();
+  }
```

Comment:
I think we should use this opportunity and introduce the mdspan names for the functions too, and then a bit closer to 5.0 deprecate the old names.